### PR TITLE
add warn when apiThunk receive not application/json response

### PR DESCRIPTION
**Pull request by sherry0429**
dash use default flask server, code in dash.py return flask.Response(mimetype='application/json')

but when developer try to use custom server, like python SimpleHttpServer, and handle request itself, when miss 'Content-Type application/json',  no warning.  developer maybe confused.

Now many python project depends on flask/django/other server, and use nginx/uwsgi etc. this pr hope reduce debug time when developer try use custom server handle dash_render.js requests.

## Contributor Checklist

- [ ✅] I have broken down my PR scope into the following TODO tasks
   -  [ ✅] add warn when apiThunk receive not application/json response
- [ ✅] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))

### CODE REVIEW

File: dash-renderer/src/actions/api.js [hunk 1]
```diff
@@ -53,6 +56,7 @@ export default function apiThunk(endpoint, method, store, id, body) {
                         return json;
                     });
                 }
+                logWarningOnce("Response miss content-type application/json")
```
Comment (alexcjohnson) [inline dash-renderer/src/actions/api.js:59]:
```suggestion
                logWarningOnce('Response is missing header: content-type: application/json');
```

Comment (sherry0429) [reply dash-renderer/src/actions/api.js]: Thanks for review

**Comment by sherry0429** [comment]
> Thank you @sherry0429 - agreed, this could save a lot of headaches! I tweaked the wording and adjusted the syntax to (I think) appease our linter. Will merge once tests pass 🎉 💃

Thanks, I see also some test mistake (lint-unit), should I still change something ?

File: dash-renderer/src/actions/api.js [hunk 2]
```diff
@@ -53,6 +56,7 @@ export default function apiThunk(endpoint, method, store, id, body) {
                         return json;
                     });
                 }
+                logWarningOnce('Response is missing header: content-type: application/json');
```
Comment (alexcjohnson) [inline dash-renderer/src/actions/api.js:59]:
```suggestion
                logWarningOnce(
                    'Response is missing header: content-type: application/json'
                );
```

**Comment by alexcjohnson** [comment]
Looks like I got it this time 😅 
In case you make another PR, there's a handy command `npm run format` (run from the `dash-renderer`) directory that will automatically fix any linting issues in the JS files.

**Comment by sherry0429** [comment]
@alexcjohnson Sorry, I have remeber it, =.=, thanks !

**Comment by alexcjohnson** [comment]
Not to worry, I forget to lint sometimes too. We appreciate your contribution!

**Comment by sherry0429** [comment]
Ok, Thanks !

**Pull request merged**